### PR TITLE
Poc web sockets

### PR DESCRIPTION
**Pull request by codyseibert**
- the work related to using websockets to get back notifications

note: I'm not sure how this will work with multiple regions... we'll need to test that out

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -100,8 +100,8 @@
     "s3-zip": "^3.1.2",
     "s3rver": "github:ustaxcourt/s3rver#created_presigned_post_support",
     "sanitize-filename": "^1.6.3",
-    "serverless": "^1.52.2",
-    "serverless-domain-manager": "^3.3.0",
+    "serverless": "^1.52.0",
+    "serverless-domain-manager": "github:codyseibert/serverless-domain-manager",
```
Comment (codyseibert) [inline package.json:104]: serverless-domain-manager does not support web sockets; therefore, we forked someone's branch that adds suppoort.

File: web-client/src/presenter/actions/destroyNotificationListenerAction.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { state } from 'cerebral';
+
+export const destroyNotificationListenerAction = async ({ get }) => {
```
Comment (rachaelparris) [inline web-client/src/presenter/actions/destroyNotificationListenerAction.js]: Is this action called somewhere? I can't find a reference to it.

**Review by rachaelparris — APPROVED** [review]

File: web-api/serverless-notifications.yml [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+  serverless-offline:
+    host: 0.0.0.0
+    port: 3020 # this port really isn't used, but needs to be unique to prevent address in use errors
+    websocketPort: 3011
+
+  alerts:
+    stages:
+      - dev
+      - stg
+      - prod
```
Comment (kkoskelin) [inline web-api/serverless-notifications.yml:27]: ` - exp` ?

File: web-api/serverless-notifications.yml [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+  masterRegion: us-east-1
+  userPoolId: us-east-1_7uRkF0Axn
+  masterDynamodbEndpoint: dynamodb.us-east-1.amazonaws.com
+  clamavDefDir: /opt/var/lib/clamav
+  deploymentBucket:
+    name: ${env:SLS_DEPLOYMENT_BUCKET}
+    serverSideEncryption: AES256
+  apiGateway:
+    binaryMediaTypes:
+      - 'application/pdf'
```
Comment (kkoskelin) [inline web-api/serverless-notifications.yml:91]: probably not needed for notifications, but for future us: this was necessary to deliver generated PDF content via API Gateway

File: shared/src/persistence/dynamo/notifications/deleteUserConnection.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    },
+    ExpressionAttributeValues: {
+      ':gsi1pk': connectionId,
+    },
+    IndexName: 'gsi1',
+    KeyConditionExpression: '#gsi1pk = :gsi1pk',
+    applicationContext,
+  });
+
+  console.log('users', users);
```
Comment (kkoskelin) [inline shared/src/persistence/dynamo/notifications/deleteUserConnection.js]: necessary?

**Review by kkoskelin — APPROVED** [review]

File: web-api/serverless-notifications.yml (hunk 3 above)
Comment (ericsorenson) [reply web-api/serverless-notifications.yml:27]: only for CloudWatch alerts

**Pull request merged**